events/moose/moose.go

```diff
@@ -54,38 +54,60 @@ type Subscriber struct {
 	currentDomain           string
 	connectionStartTime     time.Time
 	connectionToMeshnetPeer bool
-	enabled                 bool
+	consent                 config.AnalyticsConsent
 	initialHeartbeatSent    bool
 	workerCancelFunc        context.CancelFunc
 	mux                     sync.RWMutex
 }
 
+func (s *Subscriber) changeConsentState(newState config.AnalyticsConsent) error {
+	if s.consent == newState {
+		return nil
+	}
+
+	if newState == config.ConsentUndefined {
+		return fmt.Errorf("analytics consent cannot be set to and undefined state")
+	}
+
+	if s.consent == config.ConsentUndefined && newState != config.ConsentUndefined {
```

Comment:
nit: `&& newState != config.ConsentUndefined` is redundant as the function has early exit on `newState == config.ConsentUndefined`